pilot/pkg/serviceregistry/kube/controller/multicluster.go

```diff
@@ -151,29 +151,29 @@ func (m *Multicluster) close() (err error) {
 // to watch for resources being added, deleted or changed on remote clusters.
 func (m *Multicluster) ClusterAdded(cluster *multicluster.Cluster, clusterStopCh <-chan struct{}) error {
 	m.m.Lock()
-	kubeRegistry, options, configCluster, err := m.addCluster(cluster)
+	kubeController, kubeRegistry, options, configCluster, err := m.addCluster(cluster)
```

Comment:
not sure what does this do

Reply:
> not sure what does this do

By doing this, we can use this object directly instead of reading from map again in the method `initializeCluster`,  which needs to get lock.

Reply:
get it